packages/dds/map/src/directory.ts

```diff
@@ -1529,17 +1529,24 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
      */
     private createSubDirectoryCore(subdirName: string, local: boolean) {
         if (!this._subdirectories.has(subdirName)) {
-            this._subdirectories.set(
-                subdirName,
-                new SubDirectory(
-                    this.directory,
-                    this.runtime,
-                    this.serializer,
-                    posix.join(this.absolutePath, subdirName)),
-            );
+            const absolutePath = posix.join(this.absolutePath, subdirName);
+            const subDir = new SubDirectory(this.directory, this.runtime, this.serializer, absolutePath);
+            this.registerEventsOnSubDirectory(subDir, subdirName);
+            this._subdirectories.set(subdirName, subDir);
+            this.directory.emit("subDirectoryCreated", absolutePath, local, this.directory);
```

Comment:
Rather than reaching to `this.directory.emit`, it would be better for the `SharedDirectory` to watch its `root` and echo the event.  Basically the same as what we already do in the `SharedDirectory` constructor for `containedValueChanged`.

Reply:
This should also align better with the event bubbling pattern you're doing here, that the lower parts of the structure see the event before the top level.

Reply:
Sounds good. Also, what was actually the difference between "valueChanged" and "containedValueChanged" in the current structure. I see that shared directory listens to "containedValueChanged" and then emit it itself, but the shared Directory also directly emits "valueChanged" events too.

Reply:
`valueChanged` fires only on the `SharedDirectory`, and fires for any key change in the entire directory structure (this mimicked `SharedMap` to make it a drop-in replacement at the time).  `containedValueChanged` is the better event for general use though, specific to the `IDirectory`'s keys.